core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -65,13 +65,24 @@ import scala.compat.java8.OptionConverters._
 /*
  * Result metadata of a log append operation on the log
  */
-case class LogAppendResult(info: LogAppendInfo, exception: Option[Throwable] = None) {
+case class LogAppendResult(info: LogAppendInfo,
+                           exception: Option[Throwable] = None,
+                           leaderHWChange: LeaderHWChange = LeaderHWChange.None) {
   def error: Errors = exception match {
     case None => Errors.NONE
     case Some(e) => Errors.forException(e)
   }
 }
 
+/**
+ * a flag indicting whether the HWM has been changed.
+ */
+sealed trait LeaderHWChange
+object LeaderHWChange {
+  case object LeaderHWIncremented extends LeaderHWChange
```

Comment:
The usual naming convention is to only capitalize the first letter, eg LeaderHwChange.

Reply:
will roger that!